include/zephyr/data/json.h

```diff
@@ -577,14 +577,14 @@ typedef int (*json_append_bytes_t)(const char *bytes, size_t len,
  * @param len Length of JSON-encoded value
  * @param descr Pointer to the descriptor array
  * @param descr_len Number of elements in the descriptor array. Must be less
- * than 31 due to implementation detail reasons (if more fields are
+ * than 63 due to implementation detail reasons (if more fields are
  * necessary, use two descriptors)
  * @param val Pointer to the struct to hold the decoded values
  *
  * @return < 0 if error, bitmap of decoded fields on success (bit 0
  * is set if first field in the descriptor has been properly decoded, etc).
  */
-int json_obj_parse(char *json, size_t len,
+int64_t json_obj_parse(char *json, size_t len,
```

Comment:
Might consider making the bitmap a uint8_t* instead, to allow arbitrary sizes.  Returning a multi-word number by value involves significant ABI overhead on some architectures, probably more than following a pointer argument.

Reply:
The JSON parser already supports arbitrary sizes by way of calling `json_obj_parse()` multiple times. My goal with this PR was to allow more elements without increasing API complexity.

Making the API unlimited using a caller-allocated bitmap buffer is certainly possible, but it would make the API significantly more complex to use. I'd say the complexity level would be comparable to that of calling `json_obj_parse()` multiple times. And then what would we gain?

Regarding performance, as mentioned in the rationale we already return (u)int64_t in a number of Zephyr APIs, including some that can be expected to be called much more frequently than json_obj_parse().